vespalib/src/apps/vespa-detect-hostname/detect_hostname.cpp

```diff
@@ -45,6 +45,7 @@ int main(int, char **) {
             return 0;
         }
     }
-    fprintf(stderr, "ERROR: failed to detect hostname\n");
+    fprintf(stderr, "FATAL: hostname detection failed\n");
+    // XXX we should explain why it failed
```

Comment:
maybe we should not have fallbacks to localhost/127.0.0.1/::1, but simply normalize the hostname if possible and let the hostname validation tell us what was wrong with it...?